component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -201,6 +209,35 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
             if (context.isRetrying()) {
                 retryParam = "&authFailure=true&authFailureMsg=login.fail.message";
             }
+            IdentityErrorMsgContext errorContext = IdentityUtil.getIdentityErrorMsg();
+            IdentityUtil.clearIdentityErrorMsg();
+
+            String errorParam = StringUtils.EMPTY;
+            if (showAuthFailureReason) {
+                if (errorContext != null && errorContext.getErrorCode() != null) {
+                    log.debug("Identity error message context is not null.");
```

Comment:
If we are using debug logs, we need to first check whether the debug is enabled.

Reply:
Since we are not doing any string operations, `log.debug` will be enough to check and log